### PR TITLE
[New Rule] Add Jamf Protect detection rules

### PR DESCRIPTION
**Pull request by txhaflaire**
## Summary - What I changed

Added to rules using data provided by the [Jamf Protect integration](https://github.com/elastic/integrations/tree/main/packages/jamf_protect), agreed with Mika to first submit to rules and validate our approach before submitting a bunch of others.
 
 Herewith some screenshots of the rules working in Kibana.
 

image

image

image

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [ ] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? 
  - Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?
  - yes

### CODE REVIEW

**Comment by txhaflaire** [comment]
Note to reviewers: I can't add or tag any labels on this PR, maybe that's expected but hence the job fails to run.

**Comment by w0rk3r** [comment]
@txhaflaire, can we get some data samples?

**Comment by txhaflaire** [comment]
@w0rk3r Sure - how would you like to get these samples delivered and in which format?

**Comment by txhaflaire** [comment]
@w0rk3r @Mikaayenson herewith the requested data. not sure if you want me to put it into a specific folder?

[jamf_protect_events.json](https://github.com/user-attachments/files/16818028/jamf_protect_events.json)

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  ]
+]
+
+[rule]
+author = ["Thijs Xhaflaire"]
+description = """
+Identifies a high volume of `pbpaste` executions, which may indicate a bash loop continuously collecting clipboard contents, potentially allowing an attacker to harvest user credentials or other sensitive information.
+"""
+from = "now-9m"
+index = ["logs-.*"]
```
Comment (Mikaayenson) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:32]: Can you update this to be based on the integration? Should be something like `logs-jamf_protect*`

**Comment by txhaflaire** [comment]
@Mikaayenson and @w0rk3r 
[user_added_to_admin_group.csv](https://github.com/user-attachments/files/16833023/user_added_to_admin_group.csv)
[pbpaste_events.csv](https://github.com/user-attachments/files/16833024/pbpaste_events.csv)

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+]
+
+[rule]
+author = ["Thijs Xhaflaire"]
+description = """
+Identifies a high volume of `pbpaste` executions, which may indicate a bash loop continuously collecting clipboard contents, potentially allowing an attacker to harvest user credentials or other sensitive information.
+"""
+from = "now-9m"
+index = ["logs-jamf_protect*"]
+language = "kuery"
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:33]:
```suggestion
language = "eql"
```

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "threshold"
+
+query = '''
+observer.type:"Endpoint Security" and 
+observer.vendor: "Jamf" and 
+event.action:"exec" and 
+process.name:"pbpaste"
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:107]:
```suggestion
sequence by host.hostname, host.id with maxspan=1m
[process where host.os.type == "macos" and event.type == "start" and event.action == "exec"] with runs = 5
```

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
+
+[rule.threshold]
+field = ["host.hostname"]
+value = 5
+
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:129]:
```suggestion
```

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+observer.vendor: "Jamf" and 
+event.action:"exec" and 
+process.name:"pbpaste" 
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "1056"
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:114]:
```suggestion
id = "T1056"
```

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:1]: I think we can move this to the rules/macos folder

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:107]: Threshould have limited telemetry in the alert produced, we can use [EQL runs](https://www.elastic.co/guide/en/elasticsearch/reference/current/eql-syntax.html#eql-with-runs-statement) here

Comment (w0rk3r) [reply rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:107]: And, as the index is specified, we don't need to use the vendor specific fields and values

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+observer.type:"Endpoint Security" and 
+observer.vendor:"Jamf" and
+event.action:"od_group_add" and
+event.category:"configuration" and 
+event.type:"creation" and 
+group.name:"admin"
```
Comment (w0rk3r) [inline rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:105]:
```suggestion
configuration where host.os.type == "macos" and event.type == "change" and
  event.action == "od_group_add" and group.name:"admin"
```

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "query"
```
Comment (w0rk3r) [inline rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:97]:
```suggestion
type = "eql"
```

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+[rule]
+author = ["Thijs Xhaflaire"]
+description = """
+Identifies users being added to the admin group. This could be an indication of privilege
+escalation activity.
+"""
+from = "now-9m"
+index = ["logs-jamf_protect*"]
+language = "kuery"
```
Comment (w0rk3r) [inline rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:44]:
```suggestion
language = "eql"
```

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml
Comment (w0rk3r) [inline rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:1]: I think we can move this to the rules/macos folder

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Awesome stuff! Glad to see that investigation guide plugins being used, and the investigation guide itself is great.

Left a few comments around the queries, changing them to EQL as we avoid using KQL generally (Case sensitiveness, ease to tune, etc), and EQL has better telemetry than threshold rules. Let me know if you have any questions!

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml (hunk 6 above)
Comment (w0rk3r) [reply rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:105]: image

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:107]: image

**Comment by txhaflaire** [comment]
@w0rk3r just committed some suggestions. But the suggested changes removed process.name: pbpaste for that rule? Now its triggering on any exec in that timespan?

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "threshold"
+
+query = '''
+sequence by host.hostname, host.id with maxspan=1m
+[process where host.os.type == "macos" and event.type == "start" and event.action == "exec"] with runs = 5
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:103]:
```suggestion
sequence by host.hostname, host.id with maxspan=1m
[process where host.os.type == "macos" and event.type == "start" and event.action == "exec" and process.name: "pbpaste"] with runs = 5
```

**Comment by w0rk3r** [comment]
@txhaflaire Oops, sorry, just readded it to the query

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "threshold"
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:99]:
```suggestion
type = "eql"
```

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+[rule]
+author = ["Thijs Xhaflaire"]
+description = """
+Identifies a high volume of `pbpaste` executions, which may indicate a bash loop continuously collecting clipboard contents, potentially allowing an attacker to harvest user credentials or other sensitive information.
+"""
+from = "now-9m"
+index = ["logs-jamf_protect*"]
+language = "eql"
+license = "Elastic License v2"
+name = "high volume of pbpaste activity"
```
Comment (w0rk3r) [inline rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml:35]:
```suggestion
name = "Suspicious pbpaste High Volume Activity"
```

File: rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+license = "Elastic License v2"
+name = "User has been added to the Admin group"
```
Comment (w0rk3r) [inline rules/integrations/jamf/privilege_escalation_user_added_to_admin_group.toml:46]:
```suggestion
name = "User Added to the Admin Group"
```

File: rules/macos/credential_access_high_volume_of_pbpaste.toml [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.hostname, host.id with maxspan=1m
```
Comment (terrancedejesus) [inline rules/macos/credential_access_high_volume_of_pbpaste.toml:102]: Should maxspan be reduced to maybe 5s or similar? 1m seems like a large time window that could be accomplished by an end user?

**Review by terrancedejesus — APPROVED** [review]
Approving, thanks for adding these to our repository! Left some comments and reviews but ultimately seems good to go in current state.

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+observer.vendor: "Jamf" and 
+event.action:"exec" and 
+process.name:"pbpaste" 
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "1056"
```
[the remark this answers is not in the document]
Comment (terrancedejesus) [reply rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml]: Fixed.

File: rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "threshold"
+
+query = '''
+observer.type:"Endpoint Security" and
```
[the remark this answers is not in the document]
Comment (terrancedejesus) [reply rules/integrations/jamf/credential_access_high_volume_of_pbpaste.toml]: Fixed.

File: rules/macos/credential_access_high_volume_of_pbpaste.toml (hunk 13 above)
Comment (txhaflaire) [reply rules/macos/credential_access_high_volume_of_pbpaste.toml:102]: It's not a end-user scenario we're trying to capture here but a pbpaste loop that runs like once 10s to capture input and write it out to a file.

Comment (w0rk3r) [reply rules/macos/credential_access_high_volume_of_pbpaste.toml:102]: I think the question was if this is FP prone, as if the user could run the command in this 1m time window

Comment (txhaflaire) [reply rules/macos/credential_access_high_volume_of_pbpaste.toml:102]: ah right - most typical end-users don't interact with `pbpaste` at all. it's either scripts invoking `pbpaste` or `pbcopy` or in some cases software invoking it, but i've not seen cases a end-user that triggers this with that threshold set in 1 minute.

**Comment by txhaflaire** [comment]
@w0rk3r i've resolved the suggestions.

**Comment by DefSecSentinel** [comment]
Hey @txhaflaire "Suspicious pbpaste High Volume Activity" would make a great macOS detection rule but doesn't appear to be a Jamf Protect integration specific rule. It should be placed in the macOS directory "https://github.com/elastic/detection-rules/tree/main/rules/macos" and the index should be "logs-endpoint.events.*". The other rule looks specific to your alls data and integration which looks good.

**Comment by txhaflaire** [comment]
> Hey @txhaflaire "Suspicious pbpaste High Volume Activity" would make a great macOS detection rule but doesn't appear to be a Jamf Protect integration specific rule. It should be placed in the macOS directory "https://github.com/elastic/detection-rules/tree/main/rules/macos" and the index should be "logs-endpoint.events.*". The other rule looks specific to your alls data and integration which looks good. 

@DefSecSentinel do logs from the Jamf Protect integration end up in the endpoint index as well?

**Comment by w0rk3r** [comment]
> @DefSecSentinel do logs from the Jamf Protect integration end up in the endpoint index as well?

Nope, but this rule can be compatible with our agent as well, the index field is an array, we can have multiple different data sources there with no conflict

File: rules/macos/credential_access_high_volume_of_pbpaste.toml [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+  ]
+]
+
+[rule]
+author = ["Thijs Xhaflaire"]
+description = """
+Identifies a high volume of `pbpaste` executions, which may indicate a bash loop continuously collecting clipboard contents, potentially allowing an attacker to harvest user credentials or other sensitive information.
+"""
+from = "now-9m"
+index = ["logs-jamf_protect*"]
```
Comment (w0rk3r) [inline rules/macos/credential_access_high_volume_of_pbpaste.toml:32]:
```suggestion
index = ["logs-jamf_protect*", "logs-endpoint.events.process-*"]
```

File: rules/macos/credential_access_high_volume_of_pbpaste.toml [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+- Configure the integration name.
+- Click "Save and Continue".
+"""
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
```
Comment (w0rk3r) [inline rules/macos/credential_access_high_volume_of_pbpaste.toml:96]:
```suggestion
    "Data Source: Jamf Protect",
    "Data Source: Elastic Defend"
```

File: rules/macos/credential_access_high_volume_of_pbpaste.toml [hunk 18]
```diff
@@ -0,0 +1,120 @@
+[metadata]
+creation_date = "2024/08/27"
+integration = ["jamf_protect"]
```
Comment (w0rk3r) [inline rules/macos/credential_access_high_volume_of_pbpaste.toml:3]:
```suggestion
integration = ["jamf_protect", "windows"]
```

**Review by w0rk3r — APPROVED** [review]
LGTM after this minor adds, awesome work

File: rules/macos/credential_access_high_volume_of_pbpaste.toml [hunk 19]
```diff
@@ -0,0 +1,121 @@
+[metadata]
+creation_date = "2024/08/27"
+integration = ["jamf_protect", "endpoint"]
```
Comment (Mikaayenson) [inline rules/macos/credential_access_high_volume_of_pbpaste.toml:3]:
```suggestion
integration = ["endpoint"]
```

Comment (Mikaayenson) [reply rules/macos/credential_access_high_volume_of_pbpaste.toml:3]: Based on the discussion and failing unit tests. Otherwise should be g2g!

**Comment by protectionsmachine** [comment]
## Enhancement - Guidelines

These guidelines serve as a reminder set of considerations when addressing adding a new schema feature to the code.

### Documentation and Context

- [ ] Describe the feature enhancement in detail (alternative solutions, description of the solution, etc.) if not already documented in an issue.
- [ ] Include additional context or screenshots.
- [ ] Ensure the enhancement includes necessary updates to the documentation and versioning.

### Code Standards and Practices

- [ ] Code follows established design patterns within the repo and avoids duplication.
- [ ] Code changes do not introduce new warnings or errors.
- [ ] Variables and functions are well-named and descriptive.
- [ ] Any unnecessary / commented-out code is removed.
- [ ] Ensure that the code is modular and reusable where applicable.
- [ ] Check for proper exception handling and messaging.

### Testing

- [ ] New unit tests have been added to cover the enhancement.
- [ ] Existing unit tests have been updated to reflect the changes.
- [ ] Provide evidence of testing and validating the enhancement (e.g., test logs, screenshots).
- [ ] Validate that any rules affected by the enhancement are correctly updated.
- [ ] Ensure that performance is not negatively impacted by the changes.
- [ ] Verify that any release artifacts are properly generated and tested.

### Additional Schema Related Checks

- [ ] Ensure that the enhancement does not break existing functionality. (e.g., run `make test-cli`)
- [ ] Review the enhancement with a peer or team member for additional insights.
- [ ] Verify that the enhancement works across all relevant environments (e.g., different OS versions).
- [ ] Confirm that all dependencies are up-to-date and compatible with the changes.
- [ ] Link to the relevant Kibana PR or issue provided
- [ ] Exported detection rule(s) from Kibana to showcase the feature(s)
- [ ] Converted the exported ndjson file(s) to toml in the detection-rules repo
- [ ] Re-exported the toml rule(s) to ndjson and re-imported into Kibana
- [ ] Updated necessary unit tests to accommodate the feature
- [ ] Applied min_compat restrictions to limit the feature to a specified minimum stack version
- [ ] Executed all unit tests locally with a test toml rule to confirm passing
- [ ] Included Kibana PR implementer as an optional reviewer for insights on the feature
- [ ] Implemented requisite downgrade functionality
- [ ] Cross-referenced the feature with product documentation for consistency
- [ ] Incorporated a comprehensive test rule in unit tests for full schema coverage
- [ ] Conducted system testing, including fleet, import, and create APIs (e.g., run `make test-remote-cli`)

**Review by Mikaayenson — APPROVED** [review]
🚢 Great work!

**Pull request merged**